--- bin/ps/print.c	(revision 209924)
+++ bin/ps/print.c	(working copy)
@@ -431,7 +431,7 @@
 	VAR *v;
 	time_t then;
 	struct tm *tp;
-	static int use_ampm = -1;
+	static int use_time24 = -1;
 	char buf[100];
 
 	v = ve->var;
@@ -439,16 +439,16 @@
 		(void)printf("%-*s", v->width, "-");
 		return;
 	}
-	if (use_ampm < 0)
-		use_ampm = (*nl_langinfo(T_FMT_AMPM) != '\0');
+	if (use_time24 < 0)
+		use_time24 = (*nl_langinfo(T_FMT) != '\0');
 	then = k->ki_p->ki_start.tv_sec;
 	tp = localtime(&then);
 	if (now - k->ki_p->ki_start.tv_sec < 24 * 3600) {
 		(void)strftime(buf, sizeof(buf),
-		    use_ampm ? "%l:%M%p" : "%k:%M  ", tp);
+		    use_time24 ? "%H:%M" : "%l:%M%p  ", tp);
 	} else if (now - k->ki_p->ki_start.tv_sec < 7 * 86400) {
 		(void)strftime(buf, sizeof(buf),
-		    use_ampm ? "%a%I%p" : "%a%H  ", tp);
+		    use_time24 ? "%a%H" : "%a%I%p  ", tp);
 	} else
 		(void)strftime(buf, sizeof(buf), "%e%b%y", tp);
 	(void)printf("%-*s", v->width, buf);
